### DISCUSSION TITLE
No metrics available for the distination exchange when binding two exchanges

### DISCUSSION THREAD

**Discussion by AmnaSnene** [rabbitmq/rabbitmq-server#11484] [Other] [answered]
### Describe the bug

When binding two exchanges, there is no way to get the metrics of the destination exchange. Based on my understanding, this is related to the fact that no channel is publishing directly into the destination exchange.

### Reproduction steps

1. Create two exchanges (e.g., exchange_source and exchange_destination).
2. Bind exchange_source to exchange_destination.
3. Attempt to retrieve message stats for exchange_destination. (Management UI API or Prometheus)

### Expected behavior

Metrics for exchange_destination should be available and accessible, similar to other exchanges.

**Comment by michaelklishin** [comment, marked as answer]
Thank you for using RabbitMQ.

There is an explanation for this: E2E bindings are expanded when the message is published to the "original" exchange. Exchanges do not actually exist, they are just named routing tables where bindings are entries. 

Channels do the routing and when there are E2E exchanges found at routing time, they are "followed" to produce a final set of queues. No republishing happens to the 2nd layer of exchanges, so only the "first layer" goes through the code path that accounts for metrics.

There are no plans to change that. Our small team has much bigger fish to fry.

**Reply by michaelklishin** [reply]
@AmnaSnene GitHub does not send notifications for Discussion comments for at least some users, so I have to use this mention. See the above response.
